src/statement-list/statement_list_entry_point.cpp

```diff
@@ -162,46 +202,28 @@ bool statement_list_entry_point(
 
   irep_idt main_symbol_name;
 
-  // Find main symbol, if any is given.
+  // Find main symbol given by the user.
   if(config.main.has_value())
   {
-    std::list<irep_idt> matches;
-
-    for(const std::pair<const irep_idt, symbolt> &pair : symbol_table)
-      if(pair.first == config.main.value() && pair.second.type.id() == ID_code)
-        matches.push_back(pair.first);
-
-    if(matches.empty())
-    {
-      messaget message(message_handler);
-      message.error() << "main symbol `" << config.main.value() << "' not found"
-                      << messaget::eom;
+    if(search_main_symbol(symbol_table, message_handler, config.main.value()))
       return true;
-    }
-
-    if(matches.size() > 1)
-    {
-      messaget message(message_handler);
-      message.error() << "main symbol `" << config.main.value()
-                      << "' is ambiguous" << messaget::eom;
-      return true;
-    }
-
-    main_symbol_name = matches.front();
+    main_symbol_name = config.main.value();
   }
+  // Fallback: Search for block with TIA main standard name.
+  // TODO: Support the standard entry point of STL (organisation blocks).
+  // This also requires to expand the grammar and typecheck.
   else
   {
-    // TODO: Support the standard entry point of STL (organisation blocks).
-    // This also requires to expand the grammar and typecheck.
-    // For now, return false to let the typecheck itself pass (vital for
-    // --show-symbol-table). The missing entry symbol will be caught later.
-    return false;
+    if(search_main_symbol(
+         symbol_table, message_handler, ID_statement_list_main_function))
+      return true;
+    main_symbol_name = ID_statement_list_main_function;
   }
 
-  const symbolt &main = symbol_table.lookup_ref(main_symbol_name);
+  const symbolt &main_symbol = symbol_table.lookup_ref(main_symbol_name);
```

Comment:
Don't repeat type name in the variable name (unless there is a conflict): `main`